lib/membrane_rtmp_plugin/rtmp/source.ex

```diff
@@ -71,6 +109,39 @@ defmodule Membrane.RTMP.Source do
     end
   end
 
+  @impl true
+  def handle_other({:result, _result}, _ctx, state) do
+    {:ok, state}
+  end
+
+  @spec frame_provider(reference(), pid()) :: :ok
+  def frame_provider(native, target) do
+    receive do
+      :terminate ->
+        Native.stop(native)
+        :ok
+    after
+      0 ->
+        receive do
+          :get_frame ->
+            result = Native.read_frame(native)
+            send(target, {:result, result})
+
+            if result == :end_of_stream, do: :ok, else: frame_provider(native, target)
+
+          :terminate ->
+            Native.stop(native)
+            :ok
+        end
+    end
+  end
+
+  @impl true
+  def handle_playing_to_prepared(_ctx, state) do
+    send(state.provider, :terminate)
```

Comment:
Here we should call `Native.stop`

Reply:
Frame provider can be waiting for a frame right now. If we release the state, this might cause problems and is generally a race condition. I think it's better to hand the Native over to the frame provider. In fact, maybe we shouldn't even keep it in the state. Just initialize, create frame provider and forget